CalibTracker/SiStripChannelGain/src/SiStripGainsPCLHarvester.cc

```diff
@@ -139,8 +145,192 @@ void SiStripGainsPCLHarvester::dqmEndJob(DQMStore::IBooker& ibooker_, DQMStore::
   } else {
     edm::LogInfo("SiStripGainsPCLHarvester") << "Will not produce payload!" << std::endl;  
   }
+
+
+  //Collect the statistics for monitoring and validation
+  gainQualityMonitor(ibooker_,Charge_Vs_Index);
+
+}
+
+//********************************************************************************//
+void
+SiStripGainsPCLHarvester::gainQualityMonitor(DQMStore::IBooker& ibooker_, const MonitorElement* Charge_Vs_Index) const {
+  ibooker_.setCurrentFolder("AlCaReco/SiStripGainsHarvesting/");
+
+
+  std::vector<APVGain::APVmon> new_charge_histos;
+  std::vector<std::pair<std::string,std::string>> cnames = APVGain::monHnames(VChargeHisto,doChargeMonitorPerPlane,"newG2");
+  for (unsigned int i=0;i<cnames.size();i++){
+    MonitorElement* monitor = ibooker_.book1DD( (cnames[i]).first, (cnames[i]).second.c_str(), 100   , 0. , 1000. );
+    int id    = APVGain::subdetectorId((cnames[i]).first.c_str());
```

Comment:
it looks like char* was changed to string. Instead it should be const string & unless the string is changed within the function call